cvat/apps/engine/view_utils.py

```diff
@@ -92,3 +95,37 @@ def decorator(f):
         return f
 
     return decorator
+
+def get_410_response_for_export_api(path: str) -> HttpResponseGone:
+    return HttpResponseGone(textwrap.dedent(f"""\
+        This endpoint is no longer supported.
+        To initiate the export process, use POST {path}.
+        To check the process status, use GET /api/requests/rq_id,
+        where rq_id is obtained from the response of the previous request.
+        To download the prepared file, use the result_url obtained from the response of the previous request.
+    """))
+
+def get_410_response_when_checking_process_status(process_type: str, /) -> HttpResponseGone:
+    return HttpResponseGone(textwrap.dedent(f"""\
+        This endpoint no longer supports checking the status of the {process_type} process.
+        The common requests API should be used instead: GET /api/requests/rq_id,
+        where rq_id is obtained from the response of the initializing request.
+    """))
+
+class DeprecatedResponse(Response):
```

Comment:
Doesn't seem like a custom class is needed here - this could just be a function returning a `Response`.

Reply:
Are you talking about a function that accepts the same args as the `Response` constructor?  In that case, what are the benefits of such an approach? The `DeprecatedResponse` does not introduce a lot of logic, but it looks like we still work with the DRF `Response`. IMHO, `return DeprecatedResponse(msg, status=..., headers=)` looks better than `return get_deprecated_response(msg, status=..., headers=...)`

Reply:
> In that case, what are the benefits of such an approach?

It just communicates the intention better, IMO. A subclass implies some change in behavior, but in this case we essentially just change the default initialization parameters.

Having thought about this again, it actually seems to me that the best option would be to have something like this:

```
def deprecate_response(response, deprecation_date):
    response.headers["Deprecation"] = ...
    return response
```

It separates concerns nicely, and makes it clear that it just applies a modification to the response's attributes.